samples/OpenIdConnectSample/Startup.cs

```diff
@@ -39,6 +40,23 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
             app.Run(async context =>
             {
+                var authContext = new AuthenticateContext(CookieAuthenticationDefaults.AuthenticationScheme);
+                await context.Authentication.AuthenticateAsync(authContext);
+                if (context.Request.Path == new PathString("/remote-sign-out"))
+                {
+                    var session = authContext.Properties[OpenIdConnectSessionProperties.SessionState];
+                    var sid = context.Request.Query["sid"];
+                    if (string.Equals(sid, session, System.StringComparison.Ordinal))
```

Comment:
https://github.com/aspnet/Security/issues/423#issuecomment-163794309

IMHO, this comparison will unlikely work everywhere, since there's absolutely no obligation for an OP to use the same values for `sid` and `session_state`.

Identity Server, for instance, derives the session state from the session id (IIRC, it concatenates the session id and the client id and adds some salt) so you can't compare both values.

I guess it's probably not a big deal since it's just an AAD sample, but if you wanted it to be a general guidance, it would be something to fix.

/cc @leastprivilege